services/QuillLMS/app/workers/google_integration/import_classroom_students_worker.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module GoogleIntegration
+  class ImportClassroomStudentsWorker
+    include Sidekiq::Worker
+    sidekiq_options queue: SidekiqQueue::CRITICAL_EXTERNAL
+
+    PUSHER_EVENT = 'google-classroom-students-imported'
+    PUSHER_FAILED_EVENT = 'google-account-reauthorization-required'
+
+    def perform(teacher_id, selected_classroom_ids = nil)
+      teacher = ::User.find(teacher_id)
+
+      if teacher.google_authorized?
+        TeacherClassroomsStudentsImporter.run(teacher, selected_classroom_ids)
+        PusherTrigger.run(teacher_id, PUSHER_EVENT, "Google classroom students imported for #{teacher_id}.")
+      else
+        PusherTrigger.run(teacher_id, PUSHER_FAILED_EVENT, "Reauthorization needed for user #{teacher_id}.")
+      end
+    rescue => e
```

Comment:
Does this rescue prevent this retryable errors from being retried, e.g. http timeouts to google? Should we consider not rescuing here?